### ISSUE TITLE
[DatePicker] Remove onAccept prop

### ISSUE THREAD

**Issue by oliviertassinari** [mui/mui-x#4593]
The use case for the prop is described in https://github.com/mui-org/material-ui-pickers/pull/1923#issuecomment-650719129 as:

- Event: when the date is finally selected and we need to close picker".
- Use case: This case is used to make server requests based on the selected value of the date picker. When the even fires, we are sure the user has finished with the date picker.

It also seems that there are different cases when the user selects a value, like it happen with a native `<input type="date" />`:

- the user uses the keyboard, it fires a change event like, at each keystroke. Users can filter invalid dates out for form validation or making server-side requests.
- the user uses the popup, it fires a change event only once the selection of the date is completed, either on click on the day button if autoOk is true, or on click on the OK button if autoOk is false.

I come to the conclusion that the prop duplicates with `onChange`, it seems that we can drop it.

---

For a broader context, see:

- The different **states** of the date pickers: mui-org/material-ui#24150.
- The different **user flow validation** experience we can provide: https://github.com/mui-org/material-ui-pickers/issues/1927#issuecomment-658927918.

**Comment by flaviendelangle** [comment]
I am adding the `discussion` label because I am not clear about whether or not we should remove it, and the issue is quite old

**Comment by DianaKoenraadt** [comment]
What's the the difference between `onChange` and `onAccept` anyway? (Docs are useless, "onChange callback" and "Callback fired when date is accepted"... )

**Comment by flaviendelangle** [comment]
`onChange` is called whenever the date choosed is valid
`onAccept` is called at the end of the date creation process

For instance if you are on a `DateTimePicker`, `onChange` will be called when choosing the date, the hour and the minutes
Whereas `onAccept` will only be called when choosing the minutes (or when closing the picker).

Yes the doc is far from being good

**Comment by DianaKoenraadt** [comment]
Sorry, your explanation does not answer the question for me. Can you tell me when `onChange` is called and `onAccept` is not? And when is `onAccept` called but `onChange` not?

**Comment by flaviendelangle** [comment]
> For instance if you are on a DateTimePicker, onChange will be called when choosing the date, the hour and the minutes
Whereas onAccept will only be called when choosing the minutes (or when closing the picker).

I gave you an example here
I can't say when both will be called without more context, it depends on which picker you are using with which views.
In general, we call `onChange` at the end of each view and  `onAccept` on actions setting the whole date (the Today or Reset button for instance) and when closing the picker (because of a click outside or because we reached the end of the date creation process).

> I would expect that "then eof the date creation process" yields a valid date.

Yes but the opposite is not true
You have valid dates that are not at the end of the date creation process
On the DateTimePicker, when you choose the day, you have a valid date but you are not yet at the end of the date creation process so we only call `onChange`, not `onAccept`.

**Comment by DianaKoenraadt** [comment]
Ah, okay, that last sentence was the missing piece for me. Thanks!

**Comment by flaviendelangle** [comment]
Relates to #4593

**Comment by flaviendelangle** [comment]
Closing in favor of https://github.com/mui/mui-x/issues/6161
